runtime/lua/vim/lsp/buf.lua

```diff
@@ -72,7 +72,13 @@ function M.hover(config)
         if
           (
             type(result.contents) == 'table'
```

Comment:
This if condition is quite hard to follow at this point. Could we split it up into smaller/more readable blocks, or at least add a comment explaining each case that this condition checks for?

Reply:
I agree. I will give it a try, not a native lua speaker, went for the most conservative fix.

Reply:
Added comments to explain the 5 different datastructures at play here...